src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -0,0 +1,359 @@
+    IOURequestModalStack,
+    IOUBillModalStack,
+} from './ModalStacks';
+
+const Drawer = createDrawerNavigator();
+const RootStack = createStackNavigator();
+
+const defaultSubRouteOptions = {
+    cardStyle: styles.navigationScreenCardStyle,
+    headerShown: false,
+};
+
+const propTypes = {
+    network: PropTypes.shape({isOffline: PropTypes.bool}),
+    ...windowDimensionsPropTypes,
+};
+
+const defaultProps = {
+    network: {isOffline: true},
+};
+
+class AuthScreens extends React.Component {
+    constructor(props) {
+        super(props);
+
+        Timing.start(CONST.TIMING.HOMEPAGE_INITIAL_RENDER);
+        Timing.start(CONST.TIMING.HOMEPAGE_REPORTS_LOADED);
+    }
+
+    componentDidMount() {
+        NetworkConnection.listenForReconnect();
+        PusherConnectionManager.init();
+        Pusher.init({
+            appKey: CONFIG.PUSHER.APP_KEY,
+            cluster: CONFIG.PUSHER.CLUSTER,
+            authEndpoint: `${CONFIG.EXPENSIFY.URL_API_ROOT}api?command=Push_Authenticate`,
+        }).then(subscribeToReportCommentEvents);
+
+        // Fetch some data we need on initialization
+        NameValuePair.get(CONST.NVP.PRIORITY_MODE, ONYXKEYS.PRIORITY_MODE, 'default');
+        PersonalDetails.fetch();
+        PersonalDetails.fetchTimezone();
+        getBetas();
+        fetchAllReports(true, true);
+        fetchCountryCodeByRequestIP();
+        UnreadIndicatorUpdater.listenForReportChanges();
+
+        // Refresh the personal details, timezone and betas every 30 minutes
+        // There is no pusher event that sends updated personal details data yet
+        // See https://github.com/Expensify/ReactNativeChat/issues/468
+        this.interval = setInterval(() => {
+            if (this.props.network.isOffline) {
+                return;
+            }
+            PersonalDetails.fetch();
+            PersonalDetails.fetchTimezone();
+            getBetas();
+        }, 1000 * 60 * 30);
+
+        Timing.end(CONST.TIMING.HOMEPAGE_INITIAL_RENDER);
+
+        // Listen for the Command+K key being pressed so the focus can be given to the chat switcher
+        KeyboardShortcut.subscribe('K', () => {
+            redirect(ROUTES.SEARCH);
+        }, ['meta'], true);
+    }
+
+    componentWillUnmount() {
+        KeyboardShortcut.unsubscribe('K');
+        NetworkConnection.stopListeningForReconnect();
+        clearInterval(this.interval);
+        this.interval = null;
+    }
+
+    render() {
+        const modalScreenOptions = {
+            headerShown: false,
+            cardStyle: getNavigationModalCardStyle(this.props.isSmallScreenWidth),
+        };
+
+        return (
+            <RootStack.Navigator
+                mode="modal"
+            >
+                <RootStack.Screen
+                    name="Home"
+                    options={{
+                        headerShown: false,
+                        title: 'Expensify.cash',
+                    }}
+                >
+                    {() => (
```

Comment:
Good question. There is probably a way to avoid this, but in general we can pass a component to a screen via the `Component` prop or use a function as a child that returns the component. 

Probably it is more performant to do the former than the latter. However, I mostly did it here as a convenience since we are using some props available to the parent component.

Reply:
https://reactnavigation.org/docs/screen/#children

Reply:
Ok I ended up fixing this. The thing about performance in the docs got to me.